src/libraries/System.Collections.Immutable/src/System.Collections.Immutable.csproj

```diff
@@ -1,17 +1,57 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
+<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFrameworks>$(NetCoreAppCurrent);$(NetCoreAppMinimum);netstandard2.0;$(NetFrameworkMinimum)</TargetFrameworks>
     <IsPackable>true</IsPackable>
     <PackageDescription>This package provides collections that are thread safe and guaranteed to never change their contents, also known as immutable collections. Like strings, any methods that perform modifications will not change the existing instance but instead return a new instance. For efficiency reasons, the implementation uses a sharing mechanism to ensure that newly created instances share as much data as possible with the previous instance while ensuring that operations have a predictable time complexity.
 
 The System.Collections.Immutable library is built-in as part of the shared framework in .NET Runtime. The package can be installed when you need to use it in other target frameworks.</PackageDescription>
     <PackageReadmeFile>README.md</PackageReadmeFile>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
```

Comment:
without yet looking further in the implementation, can we somehow avoid this or too much perf impact?

Reply:
The unsafe? No. Even if this wasn't here, we automatically add this as well to all of our netcoreapp libraries.

At this point, AllowUnsafeBlocks is kind of meaningless. It guards use of the unsafe keyword, but with the Marshal, MemoryMarshal, and Unsafe classes, as well as DllImports, you can be just as "unsafe" even without the keyword.

Reply:
fair enough